diff --git a/dist/cjs.js b/dist/cjs.js
index 7c36cbdd06cc22c92d590be3f87a08a91ca06b40..b95c3a74970c3d61d025c7e7042af1b3ffe1caeb 100644
--- a/dist/cjs.js
+++ b/dist/cjs.js
@@ -69,6 +69,10 @@ function propertyIsOnObject(object, property) {
 	}
 }
 
+function propertyIsForbidden(key) {
+	return key === '__proto__'
+}
+
 // Protects from prototype poisoning and unexpected merging up the prototype chain.
 function propertyIsUnsafe(target, key) {
 	return propertyIsOnObject(target, key) // Properties are safe to merge if they don't exist in the target yet,
@@ -80,11 +84,14 @@ function mergeObject(target, source, options) {
 	var destination = {};
 	if (options.isMergeableObject(target)) {
 		getKeys(target).forEach(function(key) {
+			if (propertyIsForbidden(key)) {
+				return
+			}
 			destination[key] = cloneUnlessOtherwiseSpecified(target[key], options);
 		});
 	}
 	getKeys(source).forEach(function(key) {
-		if (propertyIsUnsafe(target, key)) {
+		if (propertyIsForbidden(key) || propertyIsUnsafe(target, key)) {
 			return
 		}
 
diff --git a/dist/umd.js b/dist/umd.js
index 4071e7c56ec8e5273fe166f6e723ac460fab8b72..43d9a6faf447929040d2ddd9eeec5148b5daee6e 100644
--- a/dist/umd.js
+++ b/dist/umd.js
@@ -74,6 +74,10 @@
 	}
 
 	// Protects from prototype poisoning and unexpected merging up the prototype chain.
+	function propertyIsForbidden(key) {
+		return key === '__proto__'
+	}
+
 	function propertyIsUnsafe(target, key) {
 		return propertyIsOnObject(target, key) // Properties are safe to merge if they don't exist in the target yet,
 			&& !(Object.hasOwnProperty.call(target, key) // unsafe if they exist up the prototype chain,
@@ -84,11 +88,14 @@
 		var destination = {};
 		if (options.isMergeableObject(target)) {
 			getKeys(target).forEach(function(key) {
+				if (propertyIsForbidden(key)) {
+					return
+				}
 				destination[key] = cloneUnlessOtherwiseSpecified(target[key], options);
 			});
 		}
 		getKeys(source).forEach(function(key) {
-			if (propertyIsUnsafe(target, key)) {
+			if (propertyIsForbidden(key) || propertyIsUnsafe(target, key)) {
 				return
 			}
 
diff --git a/index.js b/index.js
index 77968aed5d456cccc294f60912ef381fa1e52a7b..8ba8453c50ea5242257a877646f4b4ae18283faf 100644
--- a/index.js
+++ b/index.js
@@ -44,6 +44,10 @@ function propertyIsOnObject(object, property) {
 	}
 }
 
+function propertyIsForbidden(key) {
+	return key === '__proto__'
+}
+
 // Protects from prototype poisoning and unexpected merging up the prototype chain.
 function propertyIsUnsafe(target, key) {
 	return propertyIsOnObject(target, key) // Properties are safe to merge if they don't exist in the target yet,
@@ -55,11 +59,14 @@ function mergeObject(target, source, options) {
 	var destination = {}
 	if (options.isMergeableObject(target)) {
 		getKeys(target).forEach(function(key) {
+			if (propertyIsForbidden(key)) {
+				return
+			}
 			destination[key] = cloneUnlessOtherwiseSpecified(target[key], options)
 		})
 	}
 	getKeys(source).forEach(function(key) {
-		if (propertyIsUnsafe(target, key)) {
+		if (propertyIsForbidden(key) || propertyIsUnsafe(target, key)) {
 			return
 		}
 
